pkg/authorization/decorator.go

```diff
@@ -82,14 +83,40 @@ func (d *Decorator) AddAuditLogging() *Decorator {
 			// Note: this deviates from upstream which doesn't log audit reasons.
 			// We should rethink if this should stay.
 			logger := klog.FromContext(ctx)
-			logger.V(4).Info(auditReasonMsg)
+			logger.V(4).Info("authorization step",
+				"reason", reason,
```

Comment:
Should this not be `auditReasonMsg` ? We add this into annotations but remove from logs? Intentional?